develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown

```diff
@@ -0,0 +1,384 @@
+					</li>					
+					
+				</ul> 
+			</td>
+		</tr>
+		<tr>
+			<td class="table-header left-header">
+			MyCustomServiceUpgrade_1_0_0.java
+			</td>
+			<td class="">
+				<ul>
+					<li>
+						Disappears. It is a registration
+						on the UpgradeStepRegistrator:
+						
+						registry.register(
+						    "bundle.name","fromVersion",
+						    "toVersion",list
+						)
+					</li>					
+					
+				</ul> 
+			</td>
+		</tr>
+		<tr>
+			<td class="table-header left-header">
+			package.v1_0_0.UpgradeFoo.java
+			</td>
+			<td class="">
+				<ul>
+					<li>
+						No changes, it's already an
+						UpgradeStep.
+					</li>					
+					
+				</ul> 
+			</td>
+		</tr>
+		<tr>
+			<td class="table-header left-header">
+			package.v1_0_0.UpgradeBar.java
+			</td>
+			<td class="">
+				<ul>
+					<li>
+						No changes. Passed as a 
+						parameter in the list of
+						UpgradeSteps for the
+						registration.
+					</li>					
+					
+				</ul> 
+			</td>
+		</tr>		
+	</tbody>
+</table>
+</div>
+</div>
+</div>
+</div>
+</div>
+
+Next you can learn how to upgrade your application verify processes.
+
+## Upgrading your Verify Process to the new Framework
+
+Follow the steps below to migrate your verify process to the new framework.
+
+1.  Convert your `verify` class into a component, using the `@Component` token  
+    and define the following properties:
+    
+        @Component(
+                immediate = true,
+                property = {"verify.process.name=
+                com.liferay.[package.name.service]"},
+                service = VerifyProcess.class
+        )
+        
+    The `immediate` property specifies that the component will be available 
+    immediately, rather than the first time it is used. The 
+    `verify.process.name` property should point to the name of the service 
+    package of the app. Make sure your replace `[package.name.service]` with the 
+    service package name for your app. The OSGi service tracker uses this 
+    information to identify the verifier components. You must use 
+    `VerifyProcess.class` as the `service` property value, to denote that the
+    class is a valid implementation of the `VerifyProcess` interface.
+
+2.  Reference any Liferay services the class uses, using the `@Reference`
+    annotation. For example, here is a reference to the 
+    `dlFileVersionLocalService` for the [Document Library app](https://github.com/liferay/liferay-portal/blob/2960360870ae69360861a720136e082a06c5548f/modules/apps/collaboration/document-library/document-library-service/src/main/java/com/liferay/document/library/workflow/DLFileEntryWorkflowHandler.java)
+    :
+    
+        @Reference(unbind = "-")
+        protected void setDLFileVersionLocalService(
+          DLFileVersionLocalService dlFileVersionLocalService) {
+    
+          _dlFileVersionLocalService = dlFileVersionLocalService;
+        }
+        ...
+        private DLFileVersionLocalService _dlFileVersionLocalService;
```

Comment:
We should mention that the optional element assignment `unbind = "-"` declares that there's no unbind method.